contracts/ERC721RExample.sol

```diff
@@ -94,10 +95,14 @@ contract ERC721RExample is ERC721A, Ownable {
         Address.sendValue(payable(msg.sender), refundAmount);
     }
 
-    function refundGuaranteeActive() public view returns (bool) {
+    function getRefundGuaranteeEndTime() public view returns (uint256) {
```

Comment:
`public refundEndTime` doesn't have the same effect?

Reply:
a function can be a more explicit way to declare this specifically as part of a standard. a field is more implicit and deals with implementation details. Ideally you want the interface to not care about implementation.

Reply:
so probably what we want to do is extact this code into its own contract, and it have it implement `IERC721R`. and you're right. seems like even with ERC721 it's this way with getApproved